plugins/pdr_deterministic_plugin/ufm_sim_web_service/isolation_mgr.py

```diff
@@ -231,171 +286,121 @@ def read_next_set_of_high_ber_or_pdr_ports(self, endpoint_port):
                         continue
-
-                    raw_ber_rate, eff_ber_rate, symbol_ber_rate = self.calc_ber_rates(port_name, port_speed, port_width)
-                    if (raw_ber_rate or eff_ber_rate or symbol_ber_rate) and self.check_ber_threshold(port_name, port_speed, port_asic, fec_mode, raw_ber_rate, eff_ber_rate, symbol_ber_rate, isolation=True):
-                        issued_port = issues.get(port_name)
-                        if issued_port:
-                            issued_port.cause = Constants.ISSUE_PDR_BER
-                        else:
-                            issues[port_name] = Issue(port_name, Constants.ISSUE_BER)                    
+                    for (interval, threshold) in Constants.BER_THRESHOLDS_INTERVALS:
+                        symbol_ber_rate = self.calc_ber_rates(port_name, port_obj.active_speed, port_obj.port_width, interval)
+                        if symbol_ber_rate and symbol_ber_rate > threshold:
+                            issued_port = issues.get(port_name)
+                            if issued_port:
+                                issued_port.cause = Constants.ISSUE_PDR_BER
+                            else:
+                                issues[port_name] = Issue(port_name, Constants.ISSUE_BER)                    
+                            break                        
         return issues
 
-    def calc_single_rate(self, port_name, port_speed, port_width, col_name):
+    def calc_single_rate(self, port_name, port_speed, port_width, col_name, time_delta):
         try:
-            min_timestamp = self.ber_tele_data.loc[self.ber_tele_data[Constants.PORT_NAME] == port_name, Constants.TIMESTAMP].min()
-            max_timestamp = self.ber_tele_data.loc[self.ber_tele_data[Constants.PORT_NAME] == port_name, Constants.TIMESTAMP].max()
-            self.ber_tele_data[Constants.TIMESTAMP].min()
-            if max_timestamp == min_timestamp or max_timestamp - min_timestamp < self.ber_wait_time:
+            if port_speed != "NDR":
+                # BER calculations is only relevant for NDR
+                return 0
+            port_obj  = self.ports_data.get(port_name)
+            ber_data = port_obj.ber_tele_data
+            if not port_obj.last_symbol_ber_val or ber_data.empty:
                 return 0
-            min_value = self.ber_tele_data.loc[(self.ber_tele_data[Constants.TIMESTAMP] == min_timestamp) & (self.ber_tele_data[Constants.PORT_NAME] == port_name), col_name].values[0]
-            max_value = self.ber_tele_data.loc[(self.ber_tele_data[Constants.TIMESTAMP] == max_timestamp) & (self.ber_tele_data[Constants.PORT_NAME] == port_name), col_name].values[0]
-            if not min_value or not max_value:
+            # Calculate the adjusted time delta, rounded up to the nearest sample interval
+            adjusted_time_delta = numpy.ceil(time_delta / self.t_isolate) * self.t_isolate
```

Comment:
can you pls explain this calculation in few words

Reply:
we take two samples - the latest one, and the one that is at least an "interval" ago.